h2/src/main/org/h2/expression/Function.java

```diff
@@ -1370,12 +1370,19 @@ protected Value getValueWithArgs(Session session, Expression[] args) {
             }
             String regexpMode = v3 == null || v3.getString() == null ? "" :
                     v3.getString();
-            int flags = makeRegexpFlags(regexpMode);
+            int flags = makeRegexpFlags(regexpMode, Mode.ModeEnum.PostgreSQL.equals(database.getMode().getEnum()));
```

Comment:
`Mode.ModeEnum.PostgreSQL.equals(database.getMode().getEnum())` can be extracted into local variable.